presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2062,17 +2062,22 @@ else if (partitionUpdate.getUpdateMode() == NEW || partitionUpdate.getUpdateMode
                 // Track the manifest blob size
                 getManifestSizeInBytes(session, partitionUpdate, extraPartitionMetadata).ifPresent(hivePartitionStats::addManifestSizeInBytes);
 
+                boolean existingPartition = existingPartitions.containsKey(partitionUpdate.getName());
+                Optional<Long> existingPartitionVersion = Optional.empty();
+                if (existingPartition && partitionUpdate.getUpdateMode() == OVERWRITE) {
```

Comment:
We are doing the same check as in line 2083. Can we pull up that check to here so that we do the check only once ?

Reply:
This logic is actually is prior to building the partition using partition builder. Some part of the logic at 2083 requires partition to be built (for non overwrite case, it requires partition.getValues()), so we cannot avoid overwrite check entirely there even if we move part of the logic here, wdyt?

Reply:
we can get the partition values by calling `extractPartitionValues(partitionUpdate.getName())` from `MetastoreUtil`. We dont need the new partition object.
So i think we can move up the logic .